Sources/Citadel/SFTP/SFTPServerInboundHandler.swift

```
@@ -0,0 +1,173 @@
+                return
+            }
+            
+            let promise = context.eventLoop.makePromise(of: SFTPStatusCode.self)
+            file.close(promise: promise)
+            files[id] = nil
+            promise.futureResult.flatMap { status in
+                context.channel.writeAndFlush(
+                    SFTPMessage.status(
+                        SFTPMessage.Status(
+                            requestId: command.requestId,
+                            errorCode: status,
+                            message: "uploaded",
+                            languageTag: "EN"
+                        )
+                    )
+                )
+            }.whenFailure { _ in
+                context.channel.close(promise: nil)
+            }
+        case .read(let command):
+            print(command)
+        case .write(let command):
+            var handle = command.handle
+            
+            guard
+                let id: UInt32 = handle.readInteger(),
+                handle.readableBytes == 0,
+                let file = files[id]
+            else {
+                logger.error("bad SFTP file andle")
+                return
+            }
+            
+            let promise = context.eventLoop.makePromise(of: SFTPStatusCode.self)
+            file.write(command.data, atOffset: command.offset, promise: promise)
+            promise.futureResult.flatMap { status in
+                context.channel.writeAndFlush(
+                    SFTPMessage.status(
+                        SFTPMessage.Status(
+                            requestId: command.requestId,
+                            errorCode: status,
+                            message: "",
+                            languageTag: "EN"
+                        )
+                    )
+                )
+            }.whenFailure { _ in
+                context.channel.close(promise: nil)
+            }
+        case .handle(let command):
+            print(command)
```

Are these print statements needed? Not sure what their purpose is, or maybe replace by a log.trace? Let me know.

---

They still have to be forwarded to SFTPDelegate

---

`handle` should throw an error, and `status` and `data`, too. `mkdir` should still be forwarded to the delegate